src/data_classifier.py

```diff
@@ -0,0 +1,173 @@
+            if obj_name in stdstar.keys():
+                stdstar[obj_name].append(ob)
+            else:
+                stdstar[obj_name] = [ob]
+        
+        # all else are science targets (also consider standar star in imagety = OBJECT)
+        if imagetype == 'OBJECT':
+            if obj_name in stdstar_list:
+                # group standard obs together!
+                if obj_name in stdstar.keys():
+                    stdstar[obj_name].append(ob)
+                else:
+                    stdstar[obj_name] = [ob]
+            else:
+                # group science obs together!
+                if obj_name in science.keys():
+                    science[obj_name].append(ob)
+                else:
+                    science[obj_name] = [ob]
+
+    # #------------------
+    # science dictionay
+
+    sci_obs = []
+
+    for obj_name in science.keys():
+        obs_list = science[obj_name]
+        sci_obs.append({'sci':obs_list, 'sky':[]})
+
+
+    #------------------
+    # stdstars dictionary
+    std_obs = []
+
+    for obj_name in stdstar.keys():
+        obs_list = stdstar[obj_name]
+        std_obs.append({'sci':obs_list, 'name':obj_name,'type':['flux', 'telluric']})
+
+
+    obs_metadata = {
+        'bias' : bias,
+        'domeflat' : domeflat,
+        'twiflat' : twiflat,
+        'dark' : dark,
+        'wire' : wire,
+        'arc'  : arc,
+        'sci'  : sci_obs,
+        'std'  : std_obs}
+    
+    return obs_metadata 
+
+
+
+
+
+
+def classify(data_dir):
+
+    # What is this option for? TODO change to a default variable in the function
+    try:
+        naxis2_use = int(sys.argv[2])
+    except:
+        naxis2_use = 0
+
+    # Get list of all fits files in directory
+    all_files = os.listdir(data_dir)
+
+    # Filtering the data as per blue and red arm
+    blue_obs = []
+    red_obs = []
+
+    obs_date = None
+    for fn in all_files:
+        obs = fn.replace('.fits', '')
+
+        # Date of the observations: Is really needed?
+        if obs_date == None:
+            try:
+                f = pyfits.open(data_dir+fn)
+                obs_date = f[0].header['DATE-OBS'].split('T')[0].replace('-', '')
+                f.close()
+            except:
+                continue
+        # ------------------------------------------------
+        try:
+            f = pyfits.open(data_dir+fn)
+            camera = f[0].header['CAMERA']
+            naxis2 = f[0].header['NAXIS2']
+            f.close()
+        except:
+            continue
+        if naxis2_use!=0:
+            if naxis2 != naxis2_use:
+                continue
+        if camera == 'WiFeSBlue':
+            if obs in blue_obs:
+                continue
+            else:
+                blue_obs.append(obs)
+        if camera == 'WiFeSRed':
+            if obs in red_obs:
+                continue
+            else:
+                red_obs.append(obs)
+
+
+    blue_obs_metadata = classifier(blue_obs,data_dir)
```

Comment:
Given the variable name is `<arm>_obs_metadata`, I'd rename `classifier` to `get_obs_metadata`.